lib/srv/reexec.go

```diff
@@ -682,39 +750,44 @@ func RunForward() (errw io.Writer, code int, err error) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	var buf [1024]byte
-	var fbuf [1]*os.File
 	for {
+		var buf [1024]byte
```

Comment:
Do these need to be arrays? Why not use a slice with capacity 1024?
```suggestion
	buf := make([]byte, 0, 1024)
```

Reply:
(the two are likely to be compiled into the same thing, btw)

Reply:
Yeah, I'm not worried about the machine code, I'm worried about the Go code. A slice is generally a better abstraction than an array.

From the [Go blog](https://go.dev/blog/slices-intro):
> Arrays have their place, but they’re a bit inflexible, so you don’t see them too often in Go code. Slices, though, are everywhere. They build on arrays to provide great power and convenience.

Using a slice here would get rid of the awkward `[:]` syntax too.